core/controllers/acl_decorators_test.py

```diff
@@ -1077,6 +1168,11 @@ def setUp(self):
             self.private_exp_id, self.owner_id)
         rights_manager.publish_exploration(self.owner, self.published_exp_id)
 
+    def test_guest_can_not_delete_exploration(self):
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json(
```

Comment:
Check the error message too (for anything that's not 404).

Reply:
There is no error message for this exception. The tests with "guest" generally do not have messages as it just renders NotLoggedInException. I have added error messages for the other tests

Reply:
We probably should add an error message. Can we update base.py to provide a default error message for NotLoggedInException if there is no error message defined? Something like 'You must be logged in to access this resource.'

That way, we can have a consistent theme of having error message checks for anything that's not 404 (unless you can think of some other way to distinguish the NotLoggedIn error from the others).

Reply:
Done